kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemMetadataProvider.java

```diff
@@ -534,6 +535,16 @@ static void writeDescriptor(
         throw new DatasetIOException("Cannot close", e);
       }
     }
+
+    //rename the file into its final location so that changes appear atomic
+      try {
+          fs.rename(tempDescriptorPath, descriptorPath);
```

Comment:
Yeah, the only option I could see would potentially be going directly to the DFSClient API to call DFSClient.rename(src, dst, options).  I'm not sure what the implications of doing that would be though.  Certainly doesn't seem ideal.